graph.cpp

```diff
@@ -74,6 +75,19 @@ hessian Graph::covariance() {
   return _sol_cov;
 }
 
+void Graph::deleteFactor(int id) {
+  auto it = _factors.begin();
+  while (it != _factors.end()) {
+    if ((*it)->id == id) {
+      free(*it);
```

Comment:
It seems like the factors were created with the `new` operator so you should probably use `delete` here instead of `free()`

```suggestion
      delete *it;
```

Reply:
Good catch, I forgot about `delete`.